src/index.ts

```diff
@@ -70,19 +69,42 @@ async function serverChannel(
         // Builds initiated from the addon are always considered local
         isLocalBuild: true,
         onTaskComplete(ctx) {
-          console.log(`Completed task '${ctx.title}'`);
-          if (!announced && ctx.announcedBuild) {
-            console.debug("emitting", BUILD_ANNOUNCED, ctx.announcedBuild.id);
-            channel.emit(BUILD_ANNOUNCED, ctx.announcedBuild.id);
-            announced = true;
+          let newStep: BuildProgressPayload["step"];
+          if (step === "initialize" && ctx.announcedBuild) {
+            newStep = "build";
           }
-          if (announced && !started && ctx.build) {
-            console.debug("emitting", BUILD_STARTED, ctx.build.status);
-            channel.emit(BUILD_STARTED, ctx.build.status);
-            started = true;
+          if (step === "upload" && ctx.isolatorUrl) {
+            newStep = "verify";
           }
+          if (["build", "upload"].includes(step) && ctx.build) {
+            newStep = "snapshot";
+          }
+          if (ctx.build?.status !== "IN_PROGRESS") {
+            newStep = "complete";
+          }
```

Comment:
Not quite sure about this logic, @ghengeveld do you have a better idea?

Reply:
I'm not digging the `let newStep`. I'd rather have a function that returns the new step.

As you know I don't like `let` at all. I can sort of _let_ it slide when it's used for memory storage over time (such as `step` above), but not when it's used with a series of `if` statements to (synchronously) determine its final value.